File: clients/typescript/test/integration.test.ts
import { execSync } from "child_process";
import { BashClient } from "../src/client";

const hasBashServer = (() => {
  try {
    execSync("which bash-server", { stdio: "ignore" });
    return true;
  } catch {
    return false;
  }
})();

const describeIf = hasBashServer ? describe : describe.skip;

describeIf("Integration (requires bash-server)", () => {
  it("stdio connect + auth + eval", async () => {
    const token = "t".repeat(64);
    const client = await BashClient.connectStdio(
      "bash-server", "--stdio", "--token", token
    );
    try {
      await client.auth(token);
      const result = await client.eval("echo integration_test");
      expect(result.stdout).toContain("integration_test");
      expect(result.exit_code).toBe(0);
    } finally {
      await client.close();
    }
  });
});

File: clients/go/bashclient/integration_test.go
//go:build integration

package bashclient

import (
	"context"
	"os/exec"
	"testing"
	"time"
)

func TestIntegrationStdioConnectAuthEval(t *testing.T) {
	if _, err := exec.LookPath("bash-server"); err != nil {
		t.Skip("bash-server not found in PATH")
	}

	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
	defer cancel()

	token := "testtesttesttesttesttesttesttesttesttesttesttesttesttesttesttest"
	client, err := ConnectStdio(ctx, "bash-server", "--stdio", "--token", token)
	if err != nil {
		t.Fatal(err)
	}
	defer client.Close()

	if err := client.Auth(ctx, token); err != nil {
		t.Fatal(err)
	}

	result, err := client.Eval(ctx, "echo integration_test")
	if err != nil {
		t.Fatal(err)
	}
	if result.ExitCode != 0 {
		t.Errorf("expected exit_code=0, got %d", result.ExitCode)
	}
	if result.Stdout == "" {
		t.Error("expected non-empty stdout")
	}
}
